core/src/consensus/tendermint/mod.rs

```diff
@@ -400,23 +400,24 @@ impl Tendermint {
             Some(lock) => *vote_step > lock.on.step,
             None => true,
         };
+        let has_enough_aligned_votes = self.has_enough_aligned_votes(message);
         let lock_change = is_newer_than_lock
             && vote_step.height == self.height()
             && vote_step.step == Step::Prevote
             && message.on.block_hash.is_some()
-            && self.has_enough_aligned_votes(message);
+            && has_enough_aligned_votes;
```

Comment:
It changes the order of function calls.
Is it okay?

Reply:
It does not cause a deadlock, so I think it's okay.

Reply:
I was also worried about inconsistent data.
But it seems okay since other conditions are related to the message rather than the states of the extension.